### PR TITLE
chore(*) status poller cleanups

### PR DESCRIPTION
**Pull request by nickolaev**
### Summary

Cleanup the status poller code from the remains of the old HTTP probing mechanism. Also use DialTimeout instead.

### CODE REVIEW

File: pkg/dns/server.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ func (d *SimpleDNSServer) Start(stop <-chan struct{}) error {
 
 	errChan := make(chan error)
 	go func() {
+		defer close(errChan)
```
Comment (jakubdyszkiewicz) [inline pkg/dns/server.go:84]: Why are we closing here the channel? We are not doing it in other servers.

Comment (nickolaev) [reply pkg/dns/server.go:84]: It is a normal practice we do.

File: pkg/api-server/resource_endpoints.go [hunk 2]
```diff
@@ -20,6 +20,16 @@ import (
 
 type meshFromRequestFn = func(*restful.Request) string
 
+const (
+	k8sReadOnlyMessage = "On Kubernetes you cannot change the state of Kuma resources with 'kumactl apply' or via the HTTP API." +
+		" As a best practice, you should always be using 'kubectl apply' instead." +
+		" You can still use 'kumactl' or the HTTP API to make read-only operations. On Universal this limitation does not apply.\n"
+	//globalReadOnlyMessage = "On global control plane you can not modify dataplane resources with 'kumactl apply' or via the HTTP API." +
```
Comment (lobkovilya) [inline pkg/api-server/resource_endpoints.go:27]: Is it here intentionally?

File: pkg/api-server/server.go [hunk 3]
```diff
@@ -175,10 +175,17 @@ func (a *ApiServer) Start(stop <-chan struct{}) error {
 
 func SetupServer(rt runtime.Runtime) error {
 	cfg := rt.Config()
-	if cfg.Mode.Mode == mode.Remote {
-		for _, definition := range definitions.All {
-			if definition.ResourceFactory().GetType() != mesh.DataplaneType {
-				definition.ReadOnly = true
+	if cfg.Mode.Mode != mode.Standalone {
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/server.go:178]: I think this if is not needed if there is a switch below

File: pkg/api-server/resource_endpoints.go (hunk 2 above)
Comment (nickolaev) [reply pkg/api-server/resource_endpoints.go:27]: yes it is intentional

**Pull request merged**